lib/puppet/util/windows/adsi.rb

```diff
@@ -491,14 +488,14 @@ def self.create(name)
     def self.exists?(name_or_sid)
       well_known = false
       if (sid = Puppet::Util::Windows::SID.name_to_sid_object(name_or_sid))
-        return true if sid.account_type == 'group'
+        return true if sid[:type] == :SidTypeGroup
 
         # 'well known group' is special as it can be a group like Everyone OR a user like SYSTEM
         # so try to resolve it
         # https://msdn.microsoft.com/en-us/library/cc234477.aspx
-        well_known = sid.account_type == 'well known group'
-        return false if sid.account_type != 'alias' && !well_known
-        name_or_sid = "#{sid.domain}\\#{sid.account}"
+        well_known = sid[:type] == :SidTypeWellKnownGroup
```

Comment:
The name of this should be `account_type` instead of `type`. Correct me if I am wrong, but the call to `sid.domain` would still resolve to the hashed value `sid[:domain]` in ruby? 

If so we can ensure compat with `sid.account_type` if used elsewhere and thus why requested that it match the old name.